src/components/BattleRoom/index.js

```diff
@@ -7,55 +6,52 @@ import styled from "styled-components";
 import GameController from "../GameController";
 
 export default function BattleRoom() {
-  const params = useParams();
   const [songData, setSongData] = useState({});
-  const [roomData, setRoomData] = useState({});
   const [socket, setSocket] = useState();
```

Comment:
```suggestion
  const [socket, setSocket] = useState(null);
```
이부분 state에 초기값을 넣어줘야 할꺼같습니다.